test/WithdrawManager.test.js

```diff
@@ -0,0 +1,120 @@
+    const withdrawReceipt = await web3Child.eth.getTransactionReceipt(receipt.transactionHash)
+
+    const withdrawBlock = await web3Child.eth.getBlock(receipt.blockHash)
+    const blockHeader = getBlockHeader(withdrawBlock)
+    const headers = [blockHeader]
+    const tree = new MerkleTree(headers)
+    const root = utils.bufferToHex(tree.getRoot())
+    const start = withdrawTx.blockNumber
+    const end = withdrawTx.blockNumber
+    const withdrawBlockProof = await tree.getProof(blockHeader)
+    tree
+      .verify(
+        blockHeader,
+        withdrawBlock.number - start,
+        tree.getRoot(),
+        withdrawBlockProof
+      )
+      .should.equal(true)
+    const payload = buildSubmitHeaderBlockPaylod(accounts[0], 0, start - 1)
+    await contracts.rootChain.submitHeaderBlock(payload.vote, payload.sigs, payload.extraData)
+
+    const { vote, sigs, extraData } = buildSubmitHeaderBlockPaylod(accounts[0], start, end, root)
+    const submitHeaderBlock = await contracts.rootChain.submitHeaderBlock(vote, sigs, extraData)
+    const NewHeaderBlockEvent = submitHeaderBlock.logs.find(log => log.event == 'NewHeaderBlock')
+    const headerNumber = NewHeaderBlockEvent.args.headerBlockId
+
+    const txProof = await getTxProof(withdrawTx, withdrawBlock, web3Child)
+    assert.isTrue(verifyTxProof(txProof), 'Tx proof must be valid')
+    const receiptProof = await getReceiptProof(withdrawReceipt, withdrawBlock, web3Child)
+    assert.isTrue(
+      verifyReceiptProof(receiptProof),
+      'Receipt proof must be valid'
+    )
+
+    const burnWithdrawTx = await contracts.withdrawManager.withdrawBurntTokens(
+      headerNumber,
+      utils.bufferToHex(Buffer.concat(withdrawBlockProof)),
+      withdrawBlock.number,
+      withdrawBlock.timestamp,
+      utils.bufferToHex(withdrawBlock.transactionsRoot),
+      utils.bufferToHex(withdrawBlock.receiptsRoot),
+      utils.bufferToHex(rlp.encode(receiptProof.path)), // branch mask
+      utils.bufferToHex(getTxBytes(withdrawTx)),
+      utils.bufferToHex(rlp.encode(txProof.parentNodes)), // Merkle proof of the withdraw transaction
+      utils.bufferToHex(getReceiptBytes(withdrawReceipt)),
+      utils.bufferToHex(rlp.encode(receiptProof.parentNodes)), // Merkle proof of the withdraw receipt
+      {
+        from: user
+      }
+    )
+    // const logs = logDecoder.decodeLogs(burnWithdrawTx.receipt.rawLogs)
+    // console.log('burnWithdrawReceipt', burnWithdrawTx)
+    const log = burnWithdrawTx.logs[0]
+    log.event.should.equal('ExitStarted')
+    expect(log.args).to.include({
+      exitor: accounts[0],
+      token: childContracts.rootERC20.address
+    })
+    assertBigNumberEquality(log.args.amount, amount)
+  })
+
+  it('withdrawTokens');
+  it('withdrawDepositTokens');
+})
+
+async function deposit(depositManager, childChain, rootERC20, user, amount) {
+  await rootERC20.approve(depositManager.address, amount)
```

Comment:
we can use ERC223 so that for other tests(which needs test deposit setup) no need to approve and deposit, it'll be single TX deposit. let's discuss this.
anyway, will need it to test our fallback() deposit.

Reply:
> we can use ERC223 so that for other tests(which needs test deposit setup) no need to approve and deposit, it'll be single TX deposit. let's discuss this.
> anyway, will need it to test our fallback() deposit.